cas-server-core-services/src/main/java/org/apereo/cas/services/ServiceRegistryInitializationConfiguration.java

```diff
@@ -0,0 +1,69 @@
+package org.apereo.cas.services;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Configuration;
+
+import javax.annotation.PostConstruct;
+
+
+/**
+ * Initializes Jpa service registry database with available JSON service definitions if necessary (based on configuration flag).
+ *
+ * @author Dmitriy Kopylenko
+ * @author Misagh Moayyed
+ * @since 5.0.0
+ */
+@Configuration("serviceRegistryInitializationConfiguration")
+public class ServiceRegistryInitializationConfiguration {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ServiceRegistryInitializationConfiguration.class);
+
+    @Autowired
+    @Qualifier("serviceRegistryDao")
+    private ServiceRegistryDao serviceRegistryDao;
+
+    @Autowired
+    @Qualifier("jsonServiceRegistryDao")
+    private ServiceRegistryDao jsonServiceRegistryDao;
+
+    @Value("${svcreg.database.from.json:true}")
+    private boolean initFromJsonIfAvailable;
+
+    public ServiceRegistryInitializationConfiguration() {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Avoid unnecessary constructors - the compiler will generate these for you](https://www.codacy.com/app/mmoayyed/cas/file/2579583220/issues/source?bid=3256115&fileBranchId=3297474#l36)